### PR TITLE
implement upgrade function

### PR DESCRIPTION
**Pull request by liubo0127**

### CODE REVIEW

File: cmd/upgrade.go [hunk 1]
```diff
@@ -13,15 +13,82 @@
...
 func newUpgradeCmd() *cobra.Command {
+	opt := upgradeOptions{}
 	cmd := &cobra.Command{
-		Use:   "upgrade",
-		Short: "Upgrade a TiDB cluster",
+		Use:   "upgrade <cluster-name>",
+		Short: "Upgrade a specified TiDB cluster",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return cmd.Help()
+			if len(os.Args) != 1 {
```
Comment (AstroProfundis) [inline cmd/upgrade.go]: Just use `len(args)` here would be better.

File: cmd/upgrade.go [hunk 2]
```diff
@@ -13,15 +13,82 @@
...
+				return cmd.Help()
+			}
+			metaInfo, err := meta.ClusterMetadata(os.Args[1])
```
Comment (AstroProfundis) [inline cmd/upgrade.go]: ditto, use `args[0]`, it's the arguments parsed by conbra, and commands are already removed from the slice.

File: cmd/upgrade.go [hunk 3]
```diff
@@ -13,15 +13,82 @@
...
+
+	cmd.Flags().StringVar(&opt.version, "version", "", "version of cluster")
+	cmd.Flags().BoolVar(&opt.force, "force", false, "upgrade without transfer leader, fast but affects stability (default false)")
+	cmd.Flags().StringVar(&opt.node, "node-id", "", "node id")
+	cmd.Flags().StringVar(&opt.role, "role", "", "role name")
 	return cmd
 }
+
+func versionCompare(curVersion, newVersion string) error {
+	curVersionNotRelease, _ := regexp.MatchString("[0-9]+.*-", curVersion)
```
Comment (AstroProfundis) [inline cmd/upgrade.go]: I think it's better to check regular (release) version with `semver`, and to switch-case about special versions like `nightly`.

Comment (liubo0127) [reply cmd/upgrade.go]: Got

File: cmd/upgrade.go [hunk 4]
```diff
@@ -13,15 +13,121 @@
 
 package cmd
 
-import "github.com/spf13/cobra"
+import (
+	"fmt"
+	"github.com/pingcap-incubator/tiops/pkg/meta"
```
Comment (lonng) [inline cmd/upgrade.go]: Please separate the standard library into individual group.

Comment (AstroProfundis) [reply cmd/upgrade.go]: `go fmt` would do the trick :smiley_cat:

Comment (july2993) [reply cmd/upgrade.go]: you can use `goimports` , I think goimports will do this.

**Pull request merged**